tests: tests/drivers/build_all/eeprom : vmu_rt1170: build failure

### Describe the bug

- vmu_rt1170/mimxrt1176/cm7
tests/drivers/build_all/eeprom test build failure

ced81f6fe963c5ef2a891eb2fe887f690cf16aed is the first bad commit
commit ced81f6fe963c5ef2a891eb2fe887f690cf16aed
Author: Anthony Williams <anthony289478@gmail.com>
Date:   Thu Sep 18 17:45:15 2025 -0400

    sensors: Add missing const to sensor rtio API's
    
    Add missing const on `struct rtio_iodev *iodev`
    sensor_reconfigure_read_iodev(), sensor_stream(),
    and sensor_read(), all pass iodev to functions
    which accepts struct rtio_iodev as const.
    These functions should also be marked const.
    
    Signed-off-by: Anthony Williams <anthony289478@gmail.com>

 include/zephyr/drivers/sensor.h | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)


### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b vmu_rt1170/mimxrt1176/cm7 -T tests/drivers/build_all/eeprom
```

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | LOGS
-- | -- | -- | -- | --
drivers.eeprom.build | v4.2.0-4543-g4fe2c5bd5fe1 | Build failure | vmu_rt1170/mimxrt1176/cm7 | [142/267] Building C object zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c.c.obj[143/267] Linking C static library zephyr/drivers/i2c/libdrivers__i2c.a[144/267] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj[145/267]  Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_imx.c.obj[146/267] Linking C static library zephyr/drivers/pinctrl/libdrivers__pinctrl.a[147/267] Building C object zephyr/drivers/regulator/CMakeFiles/drivers__regulator.dir/regulator_common.c.obj[148/267]  Building C object zephyr/drivers/regulator/CMakeFiles/drivers__regulator.dir/regulator_fixed.c.obj[149/267] Linking C static library zephyr/drivers/regulator/libdrivers__regulator.a[150/267] Building C object zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.objFAILED:  zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables  -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/drivers/build_all/eeprom=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr=ZEPHYR_BASE  -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj -MF  zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj.d -o zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj -c /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.cIn file  included from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.c:81:/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/vmu_rt1170_mimxrt1176_cm7/zephyr/tests/drivers/build_all/eeprom/drivers.eeprom.build/zephyr/include/generated/zephyr/syscalls/sensor_reconfigure_read_iodev_mrsh.c:5:12:  error: conflicting types for 'z_vrfy_sensor_reconfigure_read_iodev'; have 'int(const struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *, size_t)' {aka 'int(const struct rtio_iodev *, const struct device *, const struct sensor_chan_spec  *, unsigned int)'} 5 \| extern int z_vrfy_sensor_reconfigure_read_iodev(const struct rtio_iodev * iodev, const struct device * sensor, const struct sensor_chan_spec * channels, size_t num_channels); \| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.c:71:19:  note: previous definition of 'z_vrfy_sensor_reconfigure_read_iodev' with type 'int(struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *, size_t)' {aka 'int(struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *,  unsigned int)'} 71 \| static inline int z_vrfy_sensor_reconfigure_read_iodev(struct rtio_iodev *iodev, \| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ninja: build stopped: subcommand failed.
drivers.eeprom.emul.build | v4.2.0-4543-g4fe2c5bd5fe1 | Build failure | vmu_rt1170/mimxrt1176/cm7 | [145/271] Building C object zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c.c.obj[146/271] Linking C static library zephyr/drivers/i2c/libdrivers__i2c.a[147/271] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj[148/271]  Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_imx.c.obj[149/271] Linking C static library zephyr/drivers/pinctrl/libdrivers__pinctrl.a[150/271] Building C object zephyr/drivers/regulator/CMakeFiles/drivers__regulator.dir/regulator_common.c.obj[151/271]  Building C object zephyr/drivers/regulator/CMakeFiles/drivers__regulator.dir/regulator_fixed.c.obj[152/271] Linking C static library zephyr/drivers/regulator/libdrivers__regulator.a[153/271] Building C object zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.objFAILED:  zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables  -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/drivers/build_all/eeprom=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr=ZEPHYR_BASE  -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj -MF  zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj.d -o zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj -c /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.cIn file  included from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.c:81:/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/vmu_rt1170_mimxrt1176_cm7/zephyr/tests/drivers/build_all/eeprom/drivers.eeprom.emul.build/zephyr/include/generated/zephyr/syscalls/sensor_reconfigure_read_iodev_mrsh.c:5:12:  error: conflicting types for 'z_vrfy_sensor_reconfigure_read_iodev'; have 'int(const struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *, size_t)' {aka 'int(const struct rtio_iodev *, const struct device *, const struct sensor_chan_spec  *, unsigned int)'} 5 \| extern int z_vrfy_sensor_reconfigure_read_iodev(const struct rtio_iodev * iodev, const struct device * sensor, const struct sensor_chan_spec * channels, size_t num_channels); \| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.c:71:19:  note: previous definition of 'z_vrfy_sensor_reconfigure_read_iodev' with type 'int(struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *, size_t)' {aka 'int(struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *,  unsigned int)'} 71 \| static inline int z_vrfy_sensor_reconfigure_read_iodev(struct rtio_iodev *iodev, \| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ninja: build stopped: subcommand failed.


### Relevant log output


[142/267] Building C object zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c.c.obj[143/267] Linking C static library zephyr/drivers/i2c/libdrivers__i2c.a[144/267] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj[145/267] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_imx.c.obj[146/267] Linking C static library zephyr/drivers/pinctrl/libdrivers__pinctrl.a[147/267] Building C object zephyr/drivers/regulator/CMakeFiles/drivers__regulator.dir/regulator_common.c.obj[148/267] Building C object zephyr/drivers/regulator/CMakeFiles/drivers__regulator.dir/regulator_fixed.c.obj[149/267] Linking C static library zephyr/drivers/regulator/libdrivers__regulator.a[150/267] Building C object zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.objFAILED: zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/drivers/build_all/eeprom=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj -MF zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj.d -o zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_handlers.c.obj -c /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.cIn file included from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.c:81:/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/vmu_rt1170_mimxrt1176_cm7/zephyr/tests/drivers/build_all/eeprom/drivers.eeprom.build/zephyr/include/generated/zephyr/syscalls/sensor_reconfigure_read_iodev_mrsh.c:5:12: error: conflicting types for 'z_vrfy_sensor_reconfigure_read_iodev'; have 'int(const struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *, size_t)' {aka 'int(const struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *, unsigned int)'} 5 | extern int z_vrfy_sensor_reconfigure_read_iodev(const struct rtio_iodev * iodev, const struct device * sensor, const struct sensor_chan_spec * channels, size_t num_channels); | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/sensor/sensor_handlers.c:71:19: note: previous definition of 'z_vrfy_sensor_reconfigure_read_iodev' with type 'int(struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *, size_t)' {aka 'int(struct rtio_iodev *, const struct device *, const struct sensor_chan_spec *, unsigned int)'} 71 | static inline int z_vrfy_sensor_reconfigure_read_iodev(struct rtio_iodev *iodev, | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ninja: build stopped: subcommand failed.


### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- too-chain: Zephyr
- version: v4.2.0-4543-g4fe2c5bd5fe1

### Additional Context

_No response_

--- a/drivers/sensor/sensor_handlers.c
+++ b/drivers/sensor/sensor_handlers.c
@@ -68,7 +68,7 @@ static inline int z_vrfy_sensor_get_decoder(const struct device *dev,
 }
 #include <zephyr/syscalls/sensor_get_decoder_mrsh.c>
 
-static inline int z_vrfy_sensor_reconfigure_read_iodev(struct rtio_iodev *iodev,
+static inline int z_vrfy_sensor_reconfigure_read_iodev(const struct rtio_iodev *iodev,
 						       const struct device *sensor,
 						       const struct sensor_chan_spec *channels,
 						       size_t num_channels)

